packages/rancher-monitoring/rancher-monitoring.patch

```diff
@@ -193,6 +193,15 @@ diff -x '*.tgz' -x '*.lock' -uNr packages/rancher-monitoring/charts-original/cha
      {{- end }}
      imagePullPolicy: {{ .Values.image.pullPolicy }}
    {{- if .Values.command }}
+@@ -285,7 +284,7 @@
+     {{- end }}
+     ports:
+       - name: {{ .Values.service.portName }}
+-        containerPort: {{ .Values.service.port }}
++        containerPort: {{ .Values.service.targetPort }}
```

Comment:
This is a small bug in the grafana chart; `containerPort` defines the port on which app can be reached out inside the container and `targetPort` defines the port on which container has exposed the application, so they are effectively the same thing. 

On the other hand, `port` is the port that we want to expose on the service that should lead to `targetPort`. Therefore, for Grafana we expose a port of `80`, but the `targetPort` / `containerPort` is `8080`, which is where the nginx proxy will be listening to requests to forward to the actual Grafana container running in port `3000` (unexposed).

Reply:
Aren't we exposing port 3000 right below this ?

Reply:
Yes, Grafana by default exposes both ports on https://github.com/grafana/helm-charts/blob/13382de5d9fd15f2ed28f24182c1d1fa4f4c4cba/charts/grafana/templates/_pod.tpl#L286-L292 as well.

I think this makes sense, since leaving port 3000 exposed on the deployment allows the cluster owner to use `kubectl port-forward` in order to directly access Grafana if the proxy is broken for some reason.

Reply:
So the service doesn't expose it at all but deployment does, ok.